lib/msal-browser/test/app/PublicClientApplication.spec.ts

```diff
@@ -136,6 +136,75 @@ describe("PublicClientApplication.ts Class Unit Tests", () => {
     });
 
     describe("intialize tests", () => {
+        jest.setTimeout(9999999);
+        globalThis.MessageChannel = require("worker_threads").MessageChannel; // jsdom does not include an implementation for MessageChannel
+
+        beforeEach(() => {
+            sinon.stub(MessageEvent.prototype, "source").get(() => window); // source property not set by jsdom window messaging APIs
+        });
+
+        it("serializes concurrent calls", async () => {
+            const config = {
+                auth: {
+                    clientId: TEST_CONFIG.MSAL_CLIENT_ID
+                },
+                system: {
+                    allowNativeBroker: true
+                }
+            };
+            const samples = 100;
+            const concurrency = 5;
+
+            for (let i = 1; i <= samples; i++) {
+                const app = new PublicClientApplication(config);
+                const postMessageSpy: sinon.SinonSpy = sinon.spy(window, "postMessage");
+                // @ts-ignore
+                const handshakeSpy: sinon.SinonSpy = sinon.spy(NativeMessageHandler.prototype, "sendHandshakeRequest");
+                const initEventSpy = sinon.stub(EventHandler.prototype, "emitEvent");
+                let mcPort: MessagePort;
+
+                const eventHandler = function (event: MessageEvent) {
+                    event.stopImmediatePropagation();
+                    const request = event.data;
+                    const req  = {
+                        channel: "53ee284d-920a-4b59-9d30-a60315b26836",
+                        extensionId: "test-ext-id",
```

Comment:
What about cases where the first handshake (to the specific extensionId) fails and the second handshake (without extensionId) is executed?

Reply:
Similarly, what's the behavior when one or more handshakes fail?